### PR TITLE
(Pending) Feature/GunReplacement

### PR DESCRIPTION
**Pull request by toryjleo**
- Removed all gun code
- Commented out AI gun calls
- Commented out pickups
- Commented out Arsenal

### CODE REVIEW

**Comment by Shadowmage77** [comment]
Make sure gun bulletpool does not run out of ammo!

File: Assets/Code/Scripts/GunState.cs [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+            }
+        }
+    }
+
+    /// <summary>
+    /// State for a gun that fires a burst shot
+    /// </summary>
+    public class FireBurstShot : State
+    {
+        public FireBurstShot(StateController stateController, int burstShotNum) : base(stateController)
```
Comment (toryjleo) [inline Assets/Code/Scripts/GunState.cs:171]: When timebetweenburstshot is .1f, the number of shots will go up but never exit the state

Comment (toryjleo) [reply Assets/Code/Scripts/GunState.cs:171]: Made a shotgun that had fast bursts and fast timebetweenshots

**Review by toryjleo — COMMENTED** [review]
Make sure gun bulletpool does not run out of ammo!

File: Assets/Code/Scripts/EditorObject/GunStats.cs [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+        /// </summary>
+        [SerializeField] private bool canOverheat = false;
+        /// <summary>
+        /// The amount the gun must cool down to before shooting again
+        /// </summary>
+        [Range(0, 100)] [SerializeField] private float overHeatBarrier = 50;
+        /// <summary>
+        /// How much each shot will overheat the gun
+        /// </summary>
+        [Range(1, 99)] [SerializeField] private float overHeatPercentPerShot = 5;
```
Comment (toryjleo) [inline Assets/Code/Scripts/EditorObject/GunStats.cs:89]: max of 100

File: Assets/Code/Scripts/EditorObject/GunStats.cs [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+
+        #region Overheat
+        /// <summary>
+        /// If the gun can overheat from firing
+        /// </summary>
+        [SerializeField] private bool canOverheat = false;
+        /// <summary>
+        /// The amount the gun must cool down to before shooting again
+        /// </summary>
+        [Range(0, 100)] [SerializeField] private float overHeatBarrier = 50;
```
Comment (toryjleo) [inline Assets/Code/Scripts/EditorObject/GunStats.cs:85]: coolDownBarrier

File: Assets/Code/Scripts/EditorObject/GunStats.cs [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+        public float TimeBetweenShots { get { return timeBetweenShots; } }
+        #endregion
+
+        #region Ammo
+        public bool InfiniteAmmo { get { return infiniteAmmo; } }
+        public int AmmoCount { get { return ammoCount; } }
+        #endregion
+
+        #region Multiple Projectiles
+        public int ProjectileCountPerShot { get { return projectileCountPerShot; } }
```
Comment (toryjleo) [inline Assets/Code/Scripts/EditorObject/GunStats.cs:143]: max of 40

File: Assets/Code/Editor/CustomInspector/Scripts/GunStats.cs [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+        /// Display Multiple Projectile options
+        /// </summary>
+        /// <param name="gunStats">ScriptableObject to modify</param>
+        private void MultipleProjectiles(EditorObject.GunStats gunStats)
+        {
+            EditorGUILayout.Space(5);
+
+            EditorGUILayout.LabelField("Multiple Projectiles");
+            EditorGUILayout.PropertyField(serializedObject.FindProperty("projectileCountPerShot"));
+            EditorGUILayout.PropertyField(serializedObject.FindProperty("projectileSpread"));
```
Comment (toryjleo) [inline Assets/Code/Editor/CustomInspector/Scripts/GunStats.cs:133]: make general

File: Assets/Code/Editor/CustomInspector/Scripts/GunStats.cs [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+
+            FindAndShowProperties(generalProps);
+
+            if (gunStats != null)
+            {
+                Ammunition(gunStats);
+                BurstFire(gunStats);
+                MultipleProjectiles(gunStats);
+                Overheat(gunStats);
+                BulletOptions(gunStats);
```
Comment (toryjleo) [inline Assets/Code/Editor/CustomInspector/Scripts/GunStats.cs:42]: Add a method that prints:
- Time between fire

File: Assets/Code/Editor/CustomInspector/Scripts/GunStats.cs [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+            EditorObject.GunStats gunStats = GetGunStats;
+
+            FindAndShowProperties(generalProps);
+
+            if (gunStats != null)
+            {
+                Ammunition(gunStats);
+                BurstFire(gunStats);
+                MultipleProjectiles(gunStats);
+                Overheat(gunStats);
```
Comment (toryjleo) [inline Assets/Code/Editor/CustomInspector/Scripts/GunStats.cs:41]: Print Time To Cool After Overheat

File: Assets/Code/Scripts/Bullets/BulletPool.cs [hunk 8]
```diff
@@ -5,18 +5,26 @@
...
+    /// <summary>
+    /// Initialize this class's variables. A replacement for a constructor.
+    /// </summary>
+    /// <param name="gunStats">Gun statistics to query data from</param>
     /// <param name="bulletPrefab">The template object for this pool.</param>
-    public void Init(Bullet bulletPrefab, int bulletPoolSize = 100) 
+    public void Init(EditorObject.GunStats gunStats, Bullet bulletPrefab) 
     {
-        bulletStartAmnt = bulletPoolSize;
+        bulletStartAmnt = gunStats.AmmoCount * gunStats.ProjectileCountPerShot;
```
Comment (toryjleo) [inline Assets/Code/Scripts/Bullets/BulletPool.cs:23]: cover infinite ammo case

Comment (toryjleo) [reply Assets/Code/Scripts/Bullets/BulletPool.cs:23]: Assume 200 * gunStats.ProjectileCountPerShot

**Review by Shadowmage77 — APPROVED** [review]
Big merge coming in!

**Pull request merged**